discovery/sync_manager.go

```diff
@@ -259,6 +264,7 @@ func (m *SyncManager) Stop() {
 		log.Debugf("SyncManager is stopping")
 		defer log.Debugf("SyncManager stopped")
 
+		m.cancel.WhenSome(func(fn context.CancelFunc) { fn() })
```

Comment:
To follow up [the conversation](https://github.com/lightningnetwork/lnd/pull/9680#discussion_r2032618215), if the cancel is none, it means the `Start` is never called and we can just exit?

Reply:
i think it's valuable to keep the pattern standard & simple: cause it could be the case that it was actually this very Start method that was called meaning that it did not necessarily finish & there still may be things to clean up in Stop

Reply:
cool - happy to change it to that if that's the consensus (thoughts @yyforyongyu ?) 
The reason i like the `fn.Option` here is cause we're talking about a function call-back and so it isnt super obvious that it might be nil. So the `fn.Option` makes sure you dont forget - it's also just a one liner. Totally agree that when a bunch of logic is put in the `WhenSome` then it is hard to read. 

but yeah - happy to update. let's hear what yy thinks

Reply:
> cause it could be the case that it was actually this very Start method that was called meaning that it did not necessarily finish & there still may be things to clean up in Stop

Then it's hiding a bug we should fix. There's the question that when we call `Stop`, but the `cancel` is none, the system is in an unexpected state, and we dodge the question. I think it's a broader pattern used in the repo that the state machine can result in undefined behaviors because we only do sth `WhenSome` but never act `WhenNone`. Hope we can address it someday.

Re pointer or option, kinda leaning towards option as, well, it's optional here.

Reply:
> Stop, but the cancel is none, the system is in an unexpected state, and we dodge the question

did you see reply to other comment that linked to the commit from ziggie which changed how we do Stop/Start in LND.? ie, [this commit](https://github.com/lightningnetwork/lnd/commit/653e2f36670477a64351cbe3a4aa223e6ab50107)

> Re pointer or option, kinda leaning towards option as, well, it's optional here.

👍